lib/ansible/modules/packaging/os/yum.py

```diff
@@ -742,46 +802,43 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, i
             if found:
                 continue
 
-            # if not - then pass in the spec as what to install
+            # Downgrade - The yum install command will only install or upgrade to a spec version, it will
+            # not install an older version of an RPM even if specified by the install spec. So we need to
+            # determine if this is a downgrade, and then use the yum downgrade command to install the RPM.
+            for package in pkglist:
+                # Get the NEVRA of the requested package using pkglist instead of spec because pkglist
+                #  contains consistently-formatted package names returned by yum, rather than user input
+                #  that is often not parsed correctly by splitFilename().
+                (name, ver, rel, epoch, arch) = splitFilename(package)
+
+                # Check if any version of the requested package is installed
+                inst_pkgs = is_installed(module, repoq, name, conf_file, en_repos=en_repos, dis_repos=dis_repos, is_pkg=True)
+                if inst_pkgs:
+                    (cur_name, cur_ver, cur_rel, cur_epoch, cur_arch) = splitFilename(inst_pkgs[0])
+                    compare = compareEVR((cur_epoch, cur_ver, cur_rel), (epoch, ver, rel))
+                    if compare > 0:
+                        downgrade_candidate = True
+                    else:
+                        downgrade_candidate = False
+                        break
+
+            # If package needs to be installed/upgraded/downgraded, then pass in the spec
             # we could get here if nothing provides it but that's not
             # the error we're catching here
             pkg = spec
 
         pkgs.append(pkg)
```

Comment:
Shouldn't  `pkgs.append(pkg)` be removed? As you do that again in the else clause below?